Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.ListInitExpression.cs

```diff
@@ -55,5 +56,32 @@ public NewExpression NewExpression
         return default(NewExpression);
       }
     }
+
+#if NETFRAMEWORK_4_0 || SILVERLIGHT_4_0 || SILVERLIGHT_5_0
+    //
+    // Summary:
+    //     Creates a new expression that is like this one, but using the supplied children.
+    //     If all of the children are the same, it will return this expression.
+    //
+    // Parameters:
+    //   newExpression:
+    //     The System.Linq.Expressions.ListInitExpression.NewExpression property of the
+    //     result.
+    //
+    //   initializers:
+    //     The System.Linq.Expressions.ListInitExpression.Initializers property of the result.
+    //
+    // Returns:
+    //     This expression if no children are changed or an expression with the updated
+    //     children.
+    [Pure]
+    public ListInitExpression Update(NewExpression newExpression, IEnumerable<ElementInit> initializers)
+    {
+      Contract.Requires(newExpression != null);
+      Contract.Requires(initializers != null);
+      Contract.Ensures(Contract.Result<ListInitExpression>() != null);
```

Comment:
I see another precondition here: http://referencesource.microsoft.com/#System.Core/Microsoft/Scripting/Ast/ListInitExpression.cs,231

`Contract.Requires(initializers.Any());`

Not sure that this common, but it seems that those methods are not possible to use with infinite sequences.